# Project Management

## How to set up a new support Project

TODO Describe framework and steps to set up a support project incoporating Docker, Github, command line tools, R and Nextflow

## Page setup

How to set up this page?

This page is rendered using a minimal docker container, so you have to use a specific parameter (`dev = "svglite"` in the code block definition) to display Cairo graphs.

```{r , dev = "svglite", echo = TRUE}
plot(1:10)
```

## Dropdown figure change with Vue

```{r, dev = "svglite", echo = TRUE, eval = FALSE}
library( jsonlite )
```

```{r, dev = "svglite", echo = TRUE, eval = FALSE}
plots <- data.frame(locs = list.files("./Project_1639128194_GSEA", full.names = TRUE))
plots$name <-list.files("./Project_1639128194_GSEA", full.names = FALSE)
plts <- jsonlite::toJSON(plots, dataframe = "rows", pretty=TRUE)
cat("<script> var allPlts=", plts, "</script>")
```

```{r, dev = "svglite", echo = TRUE, eval = FALSE}
<!-- include VueJS first -->
<script src="https://unpkg.com/vue@latest"></script>
<!-- use the latest vue-select release -->
<script src="https://unpkg.com/vue-select@latest"></script>
<link rel="stylesheet" href="https://unpkg.com/vue-select@latest/dist/vue-select.css">

<div id="app">
  <h1>Select a gene</h1>
  <v-select label="name" :options="options" :reduce="option => option.locs" v-model="selected" placeholder="Select a gene"></v-select>
<img :src="selected"></img>
</div>
```

```{js, dev = "svglite", echo = TRUE, eval = FALSE}
Vue.component('v-select', VueSelect.VueSelect)
new Vue({
  el: '#app',
  data: {
    selected: '',
    options: allPlts
  }
})
```

## R workflow using {Targets}

The official Targets documentation can be found [here](https://books.ropensci.org/targets/){target="blank"}.

## Glimma V2

Use glimma v2 for interactive plots of differential expression results

```{r , dev = "svglite", echo = TRUE, eval = FALSE}
### Example DESeq + Glimma run; save resulting html in Results directory
BiocManager::install("Glimma")

dge <- readRDS(
  system.file("RNAseq123/dge.rds", package = "Glimma"))
dds <- DESeq2::DESeqDataSetFromMatrix(
  countData = dge$counts,
  colData = dge$samples,
  rowData = dge$genes,
  design = ~group
)
dds <- DESeq2::DESeq(dds, quiet=TRUE)
glimmaVolcano(dds, html = "../Results/dge.html")
```

Then insert it in an iframe in the .Rmd file

```{r , dev = "svglite", echo = TRUE, eval = FALSE}
knitr::include_url("../Results/dge.html", height = "1000px")
```

## Importing WebGestaltR

How to include a WebGestaltR result webpage in a bookdown or Rmarkdown report?

First run WebGestaltR and direct the results to your dedicated Result folder

```{r , dev = "svglite", echo = TRUE, eval = FALSE}
### Example WebGestaltR run
library("WebGestaltR")

rankFile <- system.file("extdata", "GeneRankList.rnk", package="WebGestaltR")
outputDirectory <- getwd()
enrichResult <- WebGestaltR(enrichMethod="GSEA", 
                            organism="hsapiens",
                            enrichDatabase="pathway_KEGG", interestGeneFile=rankFile,
                            interestGeneType="genesymbol", 
                            sigMethod="top", 
                            topThr=10, 
                            minNum=5,
                            outputDirectory=outputDirectory)
```

Then in your report include a chunk that creates an iframe which directs to the main html filee

```{r , dev = "svglite", echo = TRUE, eval = FALSE}
knitr::include_url("../Results/Project_1638369584/Report_1638369584.html", height = "800px")
```

Then render the book as normal. 

Prior to including the url remove the header and footer with an appropriate sed command, if so desired.

```{r , dev = "svglite", echo = TRUE, eval = FALSE}
# Modify command, this one seemed to work on MacOS...
system2("sed -i '' -e '/<header>/,/<\/header>/d' Report_1639128194.html")
system2("sed -i '' -e '/<footer>/,/<\/footer>/d' Report_1639128194.html")
```



::: {.rmdnote}

Do not forget to move the resulting Report folder including the book one level up in the folder structure to preserve all figure links!

:::

## mitch

Use the mitch package for multi-contrast GSEa and interactive html report (1-20 contrasts). More infor [here](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-06856-9).


## Tips

- Create figures in PDF or TIFF so they can be manually edited later! This might be faster for small tweaks of specific plots...
- PDF images do not load in html output when viewed in Chrome, PNG works. To circumvene make figures both in png and pdf or install safari browser.

